File: Linked List/Medium/2807.Insert Greatest Common Divisors in Linked List/solution.cpp
class Solution {
public:
    ListNode* insertGreatestCommonDivisors(ListNode* head) {
        ios_base::sync_with_stdio(false); cout.tie(NULL); cin.tie(NULL);
        ListNode * temp = head;
        while (temp && temp->next)
        {
            int num1 , num2 ;
            num1 = temp->val;
            num2 = temp->next->val;
            int GCD = gcd(num1,num2);
            ListNode* node = new ListNode(GCD);
            node->next = temp->next;
            temp->next = node;
            temp = node->next;
        }
        return head;
    }
};

File: Linked List/Medium/2807.Insert Greatest Common Divisors in Linked List/solution.py
from math import gcd
from typing import Optional
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def insertGreatestCommonDivisors(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp = head 
        while temp and temp.next :
            num1 = temp.val 
            num2 = temp.next.val 
            GCD = gcd(num1,num2)
            node = ListNode(GCD)
            node.next = temp.next
            temp.next = node
            temp = node.next

        return head    